src/blockchain-utilities/derivation.ts

```diff
@@ -55,3 +55,25 @@ export const deriveBeaconId = (airnodeAddress: Address, templateId: Hex) =>
 export const deriveBeaconSetId = (beaconIds: Hex[]) =>
   // By convention beacon IDs are sorted alphabetically - the ordering impacts the resulting hash.
   ethers.utils.keccak256(ethers.utils.defaultAbiCoder.encode(['bytes32[]'], [beaconIds]));
+
+export const verifyAirnodeXpub = (airnodeXpub: string, airnodeAddress: Address): ethers.utils.HDNode => {
```

Comment:
I would kinda expect a function named "verify" to return a boolean. That said, I don't really have a better suggestion at the moment 🤷🏻‍♂️

Reply:
Yeah, but you kinda want to return the HD node. This is copied over from Airnode which is another point for preserving this as is (people are familiar and need to migrate less).